internal/controller/grafanaorganization_controller.go

```diff
@@ -131,26 +199,59 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 	}
 
 	if err != nil {
-		return ctrl.Result{}, err
+		return errors.WithStack(err)
 	}
 
 	// Update CR status if anything was changed
-	if organization.ID != grafanaOrganization.Status.OrgID {
+	if grafanaOrganization.Status.OrgID != organization.ID {
+		logger.Info("updating orgID in the grafanaOrganization status")
 		grafanaOrganization.Status.OrgID = organization.ID
 
 		if err = r.Status().Update(ctx, grafanaOrganization); err != nil {
 			logger.Error(err, "failed to update grafanaOrganization status")
-			return ctrl.Result{}, errors.WithStack(err)
+			return errors.WithStack(err)
 		}
+		logger.Info("updated orgID in the grafanaOrganization status")
 	}
 
-	// TODO add datasources for the organization.
+	return nil
+}
+
+func (r GrafanaOrganizationReconciler) configureDatasources(ctx context.Context, grafanaOrganization *v1alpha1.GrafanaOrganization) error {
+	logger := log.FromContext(ctx)
+
+	logger.Info("configuring data sources")
 
-	err = r.configureGrafana(ctx)
+	// Create or update organization in Grafana
+	var organization = grafana.Organization{
+		ID:       grafanaOrganization.Status.OrgID,
+		Name:     grafanaOrganization.Spec.DisplayName,
+		TenantID: grafanaOrganization.Name,
+	}
+
+	datasources, err := grafana.ConfigureDefaultDatasources(ctx, r.GrafanaAPI, organization)
 	if err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
+		return errors.WithStack(err)
 	}
-	return ctrl.Result{}, nil
+
+	var desiredDatasources = make([]v1alpha1.DataSource, len(datasources))
+	for i, datasource := range datasources {
+		desiredDatasources[i] = v1alpha1.DataSource{
+			ID:   datasource.ID,
+			Name: datasource.Name,
+		}
+	}
+
+	logger.Info("updating datasources in the grafanaOrganization status")
+	grafanaOrganization.Status.DataSources = desiredDatasources
```

Comment:
This should be named configuredDatasources as those were already configured and are not used as desired state.
```suggestion
	var configuredDatasources = make([]v1alpha1.DataSource, len(datasources))
	for i, datasource := range datasources {
		configuredDatasources[i] = v1alpha1.DataSource{
			ID:   datasource.ID,
			Name: datasource.Name,
		}
	}

	logger.Info("updating datasources in the grafanaOrganization status")
	grafanaOrganization.Status.DataSources = configuredDatasources
```